# Deloitte Case Study Analysis 

In [1]:
# Importing Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import datetime
%matplotlib inline

C:\Users\Vipul\Anaconda3\lib\site-packages\scipy\__init__.py:138: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.24.1)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion} is required for this version of "


In [3]:
# Importing Exchange rate file
Exchange_Rate_monthly=pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\Exchange Rate Monthly.xlsx")
Exchange_Rate_annual=pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\Exchange rate Annual.xlsx")

In [4]:
# Importing CPI Date file
CPI_Data_Annual = pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\CPI Data Annual.xlsx")
CPI_Data_Monthly = pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\CPI Data Monthly.xlsx")

In [5]:
# Importing Export Merchandise file
Export_Merchandise_Annual = pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\Exports Merchandise Annual.xlsx")
Export_Merchandise_Monthly = pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\Exports Merchandise Monthly.xlsx")
Export_Merchandise_Quarterly = pd.read_excel(r"C:\Users\Vipul\OneDrive\Documents\iNeuron\Deloitte Case Study\Exports Merchandise Quarterly.xlsx")

In [6]:
# Reading top 5 rows of Exchange_Rate_monthly
Exchange_Rate_monthly.head()

Period  Afghanistan  Angola  Albania  United Arab Emirates  Argentina  \
0  1988M12          0.0     0.0      0.0                 3.675        0.0   
1  1989M01          0.0     0.0      0.0                 3.675        0.0   
2  1989M02          0.0     0.0      0.0                 3.675        0.0   
3  1989M03          0.0     0.0      0.0                 3.675        0.0   
4  1989M04          0.0     0.0      0.0                 3.675        0.0   

   Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
0      0.0         190.3592                  0.0   1.164807  ...   
1      0.0         191.0076                  0.0   1.148810  ...   
2      0.0         191.9781                  0.0   1.169614  ...   
3      0.0         193.4470                  0.0   1.223637  ...   
4      0.0         195.0895                  0.0   1.243114  ...   

   St. Vincent and the Grenadines  Venezuela, RB  Vietnam  Vanuatu  \
0                             0.0            0.0      0.0      0.0   
1                             0.0            0.0      0.0      0.0   
2                             0.0            0.0      0.0      0.0   
3                             0.0            0.0      0.0      0.0   
4                             0.0            0.0      0.0      0.0   

   World (WBG members)  Samoa  Yemen, Rep.  South Africa  Zambia  Zimbabwe  
0             76.76546    0.0          0.0      2.349682     0.0       0.0  
1             77.67523    0.0          0.0      2.385500     0.0       0.0  
2             77.53630    0.0          0.0      2.458750     0.0       0.0  
3             78.12339    0.0          0.0      2.539391     0.0       0.0  
4             78.43185    0.0          0.0      2.548200     0.0       0.0  

[5 rows x 201 columns]

In [7]:
# Reading top 5 rows of Exchange_Rate_annual
Exchange_Rate_annual.head()

Period  Afghanistan    Angola   Albania  United Arab Emirates  Argenti0  \
0    1989          0.0     0.000    0.0000              3.675000  0.000000   
1    1990          0.0     0.000    0.0000              3.674964  0.000000   
2    1991          0.0     0.000    0.0000              3.675188  0.000000   
3    1992          0.0     0.000    0.0000              3.673874  0.991757   
4    1993          0.0  4903.561  110.0993              3.675296  0.998979   

   Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
0      0.0         213.1695              0.00000   1.262848  ...   
1      0.0         242.0275              0.00000   1.279588  ...   
2      0.0         282.4442              0.00000   1.283241  ...   
3      0.0         290.6883              0.00000   1.361321  ...   
4      0.0         299.6200              2.69682   1.471305  ...   

   St. Vincent and the Gre0dines  Venezuela, RB    Viet0m   Vanuatu  \
0                            0.0            0.0      0.00    0.0000   
1                            0.0            0.0      0.00    0.0000   
2                            0.0            0.0      0.00    0.0000   
3                            0.0            0.0      0.00    0.0000   
4                            0.0            0.0  10698.57  122.3616   

   World (WBG members)  Samoa  Yemen, Rep.  South Africa  Zambia  Zimbabwe  
0             85.26528    0.0      0.00000      2.621661     0.0       0.0  
1             90.01011    0.0      0.00000      2.587400     0.0       0.0  
2             89.66550    0.0      0.00000      2.759186     0.0       0.0  
3             91.20521    0.0      0.00000      2.853479     0.0       0.0  
4             91.83828    0.0     16.48504      3.269305     0.0       0.0  

[5 rows x 201 columns]

In [8]:
# Reading bottom 5 rows of Exchange_Rate_annual
Exchange_Rate_annual.tail()

Period  Afghanistan     Angola   Albania  United Arab Emirates   Argenti0  \
25    2014     57.24750   98.41658  105.5438              3.673000   8.120707   
26    2015     61.14346  120.09680  125.8852              3.672992   9.256635   
27    2016     67.86609  165.21090  124.1759              3.672996  14.758740   
28    2017     68.02690  167.06380  119.0652              3.673000  16.544350   
29    2018     69.36351  248.16320  107.9477              3.673000  27.165270   

     Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
25  415.9198         1188.700             2.700000   1.109586  ...   
26  477.9183         1315.008             2.700000   1.331668  ...   
27  480.4882         1315.475             2.696728   1.344980  ...   
28  482.7164         1325.904             2.683555   1.304616  ...   
29  481.9243         1394.157             2.697548   1.334079  ...   

    St. Vincent and the Gre0dines  Venezuela, RB    Viet0m    Vanuatu  \
25                            2.7       6.292221  21198.86   97.18582   
26                            2.7       6.292000  21920.18  106.68100   
27                            2.7       9.264732  22364.71  109.53110   
28                            2.7      10.040220  22713.72  107.51340   
29                            2.7   77629.490000  22992.54  107.79620   

    World (WBG members)     Samoa  Yemen, Rep.  South Africa     Zambia  \
25             438.6073  2.331331     215.0249      10.84821   6.159591   
26             495.4766  2.545019     214.9747      12.76750   8.632821   
27             527.1320  2.543874     240.7871      14.70610  10.312420   
28             567.1622  2.495690     250.2535      13.31430   9.540511   
29            1107.0490  2.534559     250.2664      13.15812  10.333910   

    Zimbabwe  
25  322.3492  
26  322.3550  
27    0.0000  
28    0.0000  
29    0.0000  

[5 rows x 201 columns]

In [15]:
# Count of all the countries, i.e rows and columns in Exchange_Rate_annual
Exchange_Rate_annual.shape

(30, 201)

### Exchange Rate Data
- About Exchange_Rate_annual Data - It shows the Exchange Rate of the Year Starting From 1989 to 2018 Country-wise which include 200 countries.
- About Exchange_Rate_Monthly Data - It shows the Exchange Rate of the Year Monthly Starting From 1989 to 2018 Country-wise which include 200 countries

In [14]:
# Reading top 5 rows of CPI_Data_Annual
CPI_Data_Annual.head()

Year  Afghanistan  Albania  United Arab Emirates   Armenia  \
0  1989          NaN      NaN                   NaN       NaN   
1  1990          NaN      NaN                   NaN       NaN   
2  1991          NaN      NaN                   NaN       NaN   
3  1992          NaN      NaN                   NaN  0.019203   
4  1993          NaN      NaN                   NaN  0.326774   

   Developing Asia  Antigua and Barbuda   Austria  Azerbaijan   Burundi  ...  \
0         35.24378                  NaN  64.49682         NaN       NaN  ...   
1         36.50166                  NaN  66.60120         NaN       NaN  ...   
2         38.06775                  NaN  68.85522         NaN       NaN  ...   
3         40.46016                  NaN  71.59559         NaN       NaN  ...   
4         45.49808                  NaN  74.24242         NaN  13.03742  ...   

     Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
0   1.036480       57.21002                             NaN       0.301840   
1   2.202740       60.29819                             NaN       0.424556   
2   4.446092       62.85179                             NaN       0.569776   
3   7.488435       64.75546                             NaN       0.748814   
4  11.539000       66.66682                             NaN       1.034275   

   Vietnam  World (WBG members)     Samoa  Yemen, Rep.  South Africa    Zambia  
0      NaN             49.71672       NaN          NaN      21.32035  0.119036  
1      NaN             52.16386       NaN          NaN      24.35065  0.222201  
2      NaN             54.75251       NaN          NaN      28.05435  0.452338  
3      NaN             56.90573       NaN          NaN      31.99856  1.174491  
4      NaN             59.44683  46.15621          NaN      35.08899  3.325080  

[5 rows x 173 columns]

In [16]:
# Reading bottom 5 rows of CPI_Data_Annual
CPI_Data_Annual.tail()

Year  Afghanistan   Albania  United Arab Emirates   Armenia  \
25  2014     140.8162  109.4389              105.7986  123.4581   
26  2015     139.8220  111.5292              110.1046  128.0211   
27  2016     145.9464  112.9599              111.8855  126.1977   
28  2017     153.2113  115.2081              114.0861  127.3832   
29  2018     154.4848  117.5585              117.9656  130.6827   

    Developing Asia  Antigua and Barbuda   Austria  Azerbaijan   Burundi  ...  \
25         121.8522             110.6661  111.2233    116.2303  148.9327  ...   
26         125.0118             111.7384  112.2428    120.9357  157.1906  ...   
27         128.5655             111.1915  113.2435    135.9781  165.9267  ...   
28         131.5174             113.8963  115.6004    153.4437  192.5622  ...   
29         134.7208                  NaN  117.4959    156.6967  190.6067  ...   

     Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
25  142.3986       109.2526                        107.6750       395.6006   
26  154.7392       109.3822                        105.8083       877.1972   
27  169.6552       110.7621                        105.6500      3113.5990   
28  180.2045       113.1215                        107.9250            NaN   
29  193.0755       115.7392                        109.9000            NaN   

     Vietnam  World (WBG members)     Samoa  Yemen, Rep.  South Africa  \
25  149.7121             115.5410  109.6602     165.6151      126.9481   
26  150.6587             120.8208  110.4576     197.8672      132.7561   
27  154.6815             137.6035  111.8962          NaN      141.1736   
28  160.1247             155.0718  113.7284          NaN      148.6171   
29  165.4669             158.8692  116.7372          NaN      154.6577   

      Zambia  
25  134.7602  
26  148.3835  
27  174.8992  
28  186.4029  
29  199.2744  

[5 rows x 173 columns]

In [17]:
# Count of all the countries, i.e rows and columns in CPI_Data_Annual
CPI_Data_Annual.shape

(30, 173)

In [18]:
# Reading top 5 rows of CPI_Data_Monthly
CPI_Data_Monthly.head()

Period  Afghanistan  Albania  United Arab Emirates  Armenia  \
0  1988M12          NaN      NaN                   NaN      NaN   
1  1989M01          NaN      NaN                   NaN      NaN   
2  1989M02          NaN      NaN                   NaN      NaN   
3  1989M03          NaN      NaN                   NaN      NaN   
4  1989M04          NaN      NaN                   NaN      NaN   

   Developing Asia  Antigua and Barbuda   Austria  Azerbaijan  Burundi  ...  \
0         33.23211                  NaN  63.07520         NaN      NaN  ...   
1         34.38592                  NaN  63.63636         NaN      NaN  ...   
2         34.60887                  NaN  63.74860         NaN      NaN  ...   
3         35.11426                  NaN  63.86083         NaN      NaN  ...   
4         35.52607                  NaN  64.08530         NaN      NaN  ...   

    Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
0  0.720415       55.61017                             NaN       0.199333   
1  0.752198       55.88706                             NaN       0.201556   
2  0.794576       56.11781                             NaN       0.207910   
3  0.858142       56.44086                             NaN       0.252126   
4  0.900519       56.81005                             NaN       0.286180   

   Vietnam  World (WBG members)  Samoa  Yemen, Rep.  South Africa    Zambia  
0      NaN             48.23800    NaN          NaN      19.62482  0.095229  
1      NaN             48.59190    NaN          NaN      19.91342  0.095229  
2      NaN             48.75023    NaN          NaN      20.20202  0.095229  
3      NaN             49.03835    NaN          NaN      20.49062  0.095229  
4      NaN             49.49379    NaN          NaN      20.77922  0.095229  

[5 rows x 173 columns]

In [19]:
# Reading bottom 5 rows of CPI_Data_Monthly
CPI_Data_Monthly.tail()

Period  Afghanistan   Albania  United Arab Emirates   Armenia  \
355  2018M07          NaN  116.6109              117.9010  127.5054   
356  2018M08          NaN  117.0569              118.0702  128.2709   
357  2018M09          NaN  117.2798              117.3614  128.6550   
358  2018M10          NaN  117.0569                   NaN       NaN   
359  2018M11          NaN       NaN                   NaN       NaN   

     Developing Asia  Antigua and Barbuda   Austria  Azerbaijan  Burundi  ...  \
355         134.7460                  NaN  117.7329    155.0722      NaN  ...   
356         135.4813                  NaN  117.7329    156.4677      NaN  ...   
357         136.0586                  NaN  118.6308    156.7807      NaN  ...   
358         136.4408                  NaN       NaN    157.0943      NaN  ...   
359              NaN                  NaN       NaN         NaN      NaN  ...   

      Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
355  195.0101       116.2995                             NaN            NaN   
356  196.3238       116.3642                             NaN            NaN   
357  197.2984       116.4994                             NaN            NaN   
358  197.7540            NaN                             NaN            NaN   
359       NaN            NaN                             NaN            NaN   

      Vietnam  World (WBG members)  Samoa  Yemen, Rep.  South Africa    Zambia  
355  165.8320             159.6118    NaN          NaN      156.5657  200.8571  
356  166.5862             160.1156    NaN          NaN      156.4214  201.4284  
357  167.5668             160.6690    NaN          NaN      157.1429  201.8189  
358  168.1249                  NaN    NaN          NaN           NaN  203.2378  
359       NaN                  NaN    NaN          NaN           NaN       NaN  

[5 rows x 173 columns]

In [20]:
# Count of all the countries, i.e rows and columns in CPI_Data_Monthly
CPI_Data_Monthly.shape

(360, 173)

### About CPI Data (Consumer Price Index) - It is used to measure Inflation
- About CPI_Data_Annual Data - It shows the CPI Data of the Year Starting From 1989 to 2018 Country-wise which include 172 countries.
- About CPI_Data_Monthly Data - It shows the CPI Data of the Year Monthly Starting From 1989 to 2018 Country-wise which include 172 countries

In [21]:
# Reading top 5 rows of Export_Merchandise_Annual
Export_Merchandise_Annual.head()

Year  Argentina  Developing Asia  Australia   Belgium    Brazil  Canada  \
0  1991   22191.58              NaN   80560.51       NaN  53953.57     NaN   
1  1992   22254.20              NaN   85549.90       NaN  63269.99     NaN   
2  1993   24153.33              NaN   90912.32       NaN  73589.75     NaN   
3  1994   28583.86              NaN   96492.75       NaN  75080.52     NaN   
4  1995   35332.53              NaN   99382.24  296008.8  70497.63     NaN   

   Chile  China  Czech Republic  ...  Singapore  \
0    NaN    NaN             NaN  ...   47745.83   
1    NaN    NaN             NaN  ...   51734.70   
2    NaN    NaN             NaN  ...   61247.96   
3    NaN    NaN             NaN  ...   78650.06   
4    NaN    NaN             NaN  ...   91012.18   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand     Turkey  \
0                            NaN       NaN  66384.82  44867.88   51.61588   
1                            NaN       NaN  66534.56  50383.36   89.88207   
2                            NaN       NaN  72278.16  57262.15  156.75590   
3                            NaN       NaN  83427.53  66941.26  521.16430   
4                            NaN       NaN  90334.67  76568.04  837.45090   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       64603.98     NaN       528704.3                  NaN  
1       68509.85     NaN       561466.6                  NaN  
2       71342.15     NaN       579401.9                  NaN  
3       77694.99     NaN       625109.1            4948934.0  
4       87338.68     NaN       679022.4            5375153.0  

[5 rows x 55 columns]

In [22]:
# Reading bottom 5 rows of Export_Merchandise_Annual
Export_Merchandise_Annual.tail()

Year  Argentina  Developing Asia  Australia   Belgium    Brazil    Canada  \
23  2014   58352.33        2563743.0   206921.6  374254.8  191186.3  466.2965   
24  2015   57395.60        2571950.0   213902.5  385915.2  207865.9  467.9437   
25  2016   61266.24        2571423.0   228136.4  364903.6  214051.1  470.2580   
26  2017   61013.20        2738630.0   228618.9  357317.1  228728.2  479.5822   
27  2018        NaN              NaN        NaN       NaN       NaN       NaN   

       Chile      China  Czech Republic  ...  Singapore  \
23  73861.92  1919820.0        180054.9  ...   393277.9   
24  72896.57  1914839.0        196038.9  ...   394916.2   
25  72660.40  1912261.0        207237.7  ...   395946.5   
26  71263.73  2015050.0        221517.8  ...   424983.8   
27       NaN        NaN             NaN  ...        NaN   

    Sub-Saharan Africa developing   Slovakia    Sweden  Thailand    Turkey  \
23                       3065.586   91.58199  164751.0  212844.8  220984.1   
24                       4282.243   97.57113  170620.6  209078.3  278274.8   
25                       4995.761  103.20390  175780.5  210790.6  320703.7   
26                       5466.110  107.31260  182405.5  223469.9  420187.2   
27                            NaN        NaN       NaN       NaN       NaN   

    Taiwan, China    Uganda  United States  World (WBG members)  
23       290097.6  3065.586      1480185.0           13082523.0  
24       283191.5  4282.243      1464425.0           13487933.0  
25       288395.4  4995.761      1460203.0           13774844.0  
26       313299.8  5466.110      1519418.0           14556736.0  
27            NaN       NaN            NaN                  NaN  

[5 rows x 55 columns]

In [23]:
# Count of all the countries, i.e rows and columns in Export_Merchandise_Annual
Export_Merchandise_Annual.shape

(28, 55)

In [24]:
# Reading top 5 rows of Export_Merchandise_Monthly
Export_Merchandise_Monthly.head()

Period  Argentina  Developing Asia  Australia  Belgium    Brazil  Canada  \
0  1991M01   1265.303         3313.754   5502.346      NaN  4896.404     NaN   
1  1991M02   1518.856              NaN   5708.852      NaN  3887.668     NaN   
2  1991M03   1632.372              NaN   6890.402      NaN  4384.693     NaN   
3  1991M04   1884.038              NaN   6524.655      NaN  5119.613     NaN   
4  1991M05   2302.835              NaN   7145.694      NaN  4917.469     NaN   

   Chile  China  Czech Republic  ...  Singapore  \
0    NaN    NaN             NaN  ...   4014.566   
1    NaN    NaN             NaN  ...   3175.083   
2    NaN    NaN             NaN  ...   4062.258   
3    NaN    NaN             NaN  ...   3896.417   
4    NaN    NaN             NaN  ...   3908.034   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand    Turkey  \
0                            NaN       NaN  5493.817  3313.754  2.651407   
1                            NaN       NaN  5402.757  3152.839  2.588754   
2                            NaN       NaN  5898.220  4010.966  3.935234   
3                            NaN       NaN  6062.707  3425.034  3.178725   
4                            NaN       NaN  5544.209  3584.901  3.911652   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       5268.258     NaN       41353.47             168083.9  
1       4161.086     NaN       40909.06                  NaN  
2       4857.343     NaN       46101.14                  NaN  
3       4778.434     NaN       45152.55                  NaN  
4       5758.457     NaN       45104.24                  NaN  

[5 rows x 55 columns]

In [25]:
# Reading bottom 5 rows of Export_Merchandise_Monthly
Export_Merchandise_Monthly.tail()

Period  Argentina  Developing Asia  Australia   Belgium    Brazil  \
328  2018M05   4816.634              NaN   20025.57  28613.68  19284.53   
329  2018M06   4678.404              NaN   21514.23  28093.34  20220.21   
330  2018M07        NaN              NaN        NaN       NaN  22464.32   
331  2018M08        NaN              NaN        NaN       NaN       NaN   
332  2018M09        NaN              NaN        NaN       NaN       NaN   

       Canada     Chile  China  Czech Republic  ...  Singapore  \
328  41.77341  6488.860    NaN        19095.95  ...   38430.16   
329  43.83378  6309.378    NaN        19538.32  ...   36101.53   
330       NaN       NaN    NaN        17102.93  ...   37948.72   
331       NaN       NaN    NaN             NaN  ...        NaN   
332       NaN       NaN    NaN             NaN  ...        NaN   

     Sub-Saharan Africa developing  Slovakia    Sweden  Thailand    Turkey  \
328                            NaN  9.767317  16014.38  20213.18  44712.29   
329                            NaN  9.919567  16035.70  19859.37  42321.98   
330                            NaN       NaN  13735.09  18668.80  48067.83   
331                            NaN       NaN       NaN       NaN       NaN   
332                            NaN       NaN       NaN       NaN       NaN   

     Taiwan, China  Uganda  United States  World (WBG members)  
328       27710.60     NaN       136559.9            1233690.0  
329       26755.98     NaN       136749.6            1246063.0  
330       27016.64     NaN       126438.5            1227797.0  
331       27080.79     NaN            NaN                  NaN  
332            NaN     NaN            NaN                  NaN  

[5 rows x 55 columns]

In [26]:
# Count of all the countries, i.e rows and columns in Export_Merchandise_Monthly
Export_Merchandise_Monthly.shape

(333, 55)

In [27]:
# Reading top 5 rows of Export_Merchandise_Quarterly
Export_Merchandise_Quarterly.head()

Period  Argentina  Developing Asia  Australia  Belgium    Brazil  Canada  \
0  1991Q1   4416.531              NaN   18101.60      NaN  13168.76     NaN   
1  1991Q2   6318.661              NaN   20387.41      NaN  14826.59     NaN   
2  1991Q3   6371.975              NaN   20408.96      NaN  12690.13     NaN   
3  1991Q4   5084.412              NaN   21662.54      NaN  13268.09     NaN   
4  1992Q1   4724.392              NaN   19736.18      NaN  13636.11     NaN   

   Chile  China  Czech Republic  ...  Singapore  \
0    NaN    NaN             NaN  ...   11251.91   
1    NaN    NaN             NaN  ...   12001.44   
2    NaN    NaN             NaN  ...   12429.65   
3    NaN    NaN             NaN  ...   12062.84   
4    NaN    NaN             NaN  ...   11953.98   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand     Turkey  \
0                            NaN       NaN  16794.79  10477.56   9.175394   
1                            NaN       NaN  17536.09  10505.70  10.675940   
2                            NaN       NaN  14657.51  12132.41  13.381980   
3                            NaN       NaN  17396.43  11752.21  18.382570   
4                            NaN       NaN  17557.24  11931.11  17.588660   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       14286.69     NaN       128363.7                  NaN  
1       16374.93     NaN       134818.8                  NaN  
2       17147.92     NaN       127887.5                  NaN  
3       16794.44     NaN       137634.3                  NaN  
4       16101.55     NaN       139734.4                  NaN  

[5 rows x 55 columns]

In [28]:
# Reading bottom 5 rows of Export_Merchandise_Quarterly
Export_Merchandise_Quarterly.tail()

Period  Argentina  Developing Asia  Australia   Belgium    Brazil  \
106  2017Q3   16770.97         720194.3   59437.88  90699.26  61736.42   
107  2017Q4   14745.50         735916.6   58454.59  88748.78  55125.52   
108  2018Q1   14262.01              NaN   56989.81  85198.43  54994.27   
109  2018Q2   14454.23              NaN   60143.25  84912.30  59141.24   
110  2018Q3        NaN              NaN        NaN       NaN       NaN   

       Canada     Chile     China  Czech Republic  ...  Singapore  \
106  115.8469  17700.18  531477.9        52637.89  ...   107666.9   
107  121.1300  18825.41  552142.0        57857.50  ...   110265.1   
108  116.7996  19143.87       NaN        57096.06  ...   105758.2   
109  126.2617  18995.88       NaN        57455.02  ...   110839.1   
110       NaN       NaN       NaN             NaN  ...        NaN   

     Sub-Saharan Africa developing  Slovakia    Sweden  Thailand     Turkey  \
106                       1290.058  25.48270  43234.80  58368.11   95786.19   
107                       1467.184  28.52851  48239.38  56796.56  114474.60   
108                       1569.471  27.04031  46676.40  57476.45  109945.10   
109                            NaN  28.77668  47477.92  57318.32  126211.00   
110                            NaN       NaN       NaN       NaN        NaN   

     Taiwan, China    Uganda  United States  World (WBG members)  
106       82612.97  1290.058       374427.1            3631192.0  
107       83571.29  1467.184       395461.9            3810925.0  
108       76497.67  1569.471       385749.9            3619569.0  
109       79850.02       NaN       404231.9            3682867.0  
110            NaN       NaN            NaN                  NaN  

[5 rows x 55 columns]

In [29]:
# Count of all the countries, i.e rows and columns in Export_Merchandise_Quarterly
Export_Merchandise_Quarterly.shape

(111, 55)

### About Export Merchandise Data
- About Export_Merchandise_Annual Data - It shows the Export Merchandise Data of the Year Starting From 1991 to 2018 Country-wise which include 54 countries.
- About Export_Merchandise_Monthly Data - It shows the Export Merchandise Data of the Year Monthly Starting From 1991 to 2018 Country-wise which include 54 countries
- About Export_Merchandise_Quarterly Data - It shows the Export Merchandise Data of the Year Quarterly Starting From 1991 to 2018 Country-wise which include 54 countries

# Checking for the Missing Data or Null Values
#### In this we will check for Null values. Replace the Null values with the appropriate values in all the above datasets

In [31]:
Exchange_Rate_monthly.isnull().sum()

Period                  0
Afghanistan             0
Angola                  0
Albania                 0
United Arab Emirates    0
                       ..
Samoa                   0
Yemen, Rep.             0
South Africa            0
Zambia                  0
Zimbabwe                0
Length: 201, dtype: int64

In [32]:
Exchange_Rate_annual.isnull().sum()

Period                  0
Afghanistan             0
Angola                  0
Albania                 0
United Arab Emirates    0
                       ..
Samoa                   0
Yemen, Rep.             0
South Africa            0
Zambia                  0
Zimbabwe                0
Length: 201, dtype: int64

#### For the above Dataset of Exchange Rate we observed there is not any Null values 

In [33]:
CPI_Data_Annual.isnull().sum()

Year                     0
Afghanistan             15
Albania                  5
United Arab Emirates    18
Armenia                  3
                        ..
World (WBG members)      0
Samoa                    4
Yemen, Rep.             15
South Africa             0
Zambia                   0
Length: 173, dtype: int64

In [35]:
# This will give the sum of all the null values present in the CPI_Data_Annual
CPI_Data_Annual.isnull().sum().sum()

910

In [34]:
CPI_Data_Monthly.isnull().sum()

Period                    0
Afghanistan             190
Albania                  62
United Arab Emirates    219
Armenia                  50
                       ... 
World (WBG members)       2
Samoa                    58
Yemen, Rep.             180
South Africa              2
Zambia                    1
Length: 173, dtype: int64

In [36]:
# This will give the sum of all the null values present in the CPI_Data_Monthly
CPI_Data_Monthly.isnull().sum().sum()

11594

#### For the above Dataset of CPI_Data we observed there is Null values present in both the tables
- In CPI_Data_Annual there is total 910 Null Values
- In CPI_Data_Monthly there is total 11594 Null Values

#### So First, We will try to fix this. We need to handle these missing values, so that our data is Complete , cleaned and Structured. By Filling missing values with Zero un both the tables

In [46]:
# Filling Null Values with zeroes:
CPI_Data_Monthly=CPI_Data_Monthly.fillna(0)
CPI_Data_Annual=CPI_Data_Annual.fillna(0)

In [38]:
CPI_Data_Monthly.isnull().sum()

Period                  0
Afghanistan             0
Albania                 0
United Arab Emirates    0
Armenia                 0
                       ..
World (WBG members)     0
Samoa                   0
Yemen, Rep.             0
South Africa            0
Zambia                  0
Length: 173, dtype: int64

In [39]:
CPI_Data_Annual.isnull().sum()

Year                    0
Afghanistan             0
Albania                 0
United Arab Emirates    0
Armenia                 0
                       ..
World (WBG members)     0
Samoa                   0
Yemen, Rep.             0
South Africa            0
Zambia                  0
Length: 173, dtype: int64

#### Now the CPI Data is clean and we have replace all the null values with zeros

In [40]:
Export_Merchandise_Annual.isnull().sum()

Year                                     0
Argentina                                1
Developing Asia                         16
Australia                                1
Belgium                                  5
Brazil                                   1
Canada                                   7
Chile                                   13
China                                   15
Czech Republic                           8
Germany                                  1
Developing Countries                    16
Denmark                                  5
East Asia & Pacific developing          16
Europe & Central Asia developing        17
Spain                                    1
Estonia                                  4
Finland                                  5
France                                   9
United Kingdom                           3
High Income Countries                    1
Hong Kong SAR, China                     1
Hungary                                 14
High income

In [41]:
Export_Merchandise_Annual.isnull().sum().sum()

368

In [42]:
Export_Merchandise_Monthly.isnull().sum()

Period                                    0
Argentina                                 3
Developing Asia                         175
Australia                                 3
Belgium                                  51
Brazil                                    2
Canada                                   75
Chile                                   147
China                                   175
Czech Republic                           86
Germany                                   2
Developing Countries                    175
Denmark                                  50
East Asia & Pacific developing          175
Europe & Central Asia developing        183
Spain                                     3
Estonia                                  38
Finland                                  50
France                                   98
United Kingdom                           30
High Income Countries                     2
Hong Kong SAR, China                      2
Hungary                         

In [43]:
Export_Merchandise_Monthly.isnull().sum().sum()

3821

In [44]:
Export_Merchandise_Quarterly.isnull().sum()

Period                                    0
Argentina                                 1
Developing Asia                          60
Australia                                 1
Belgium                                  17
Brazil                                    1
Canada                                   25
Chile                                    49
China                                    59
Czech Republic                           29
Germany                                   1
Developing Countries                     60
Denmark                                  17
East Asia & Pacific developing           60
Europe & Central Asia developing         62
Spain                                     1
Estonia                                  13
Finland                                  17
France                                   33
United Kingdom                           10
High Income Countries                     1
Hong Kong SAR, China                      1
Hungary                         

In [45]:
Export_Merchandise_Quarterly.isnull().sum().sum()

1298

#### For the above Dataset of Export Merchandise Data we observed there is Null values present in all the tables.
- In Export_Merchandise_Annual there is total 368 Null Values.
- In Export_Merchandise_Monthly there is total 3281 Null Values.
- In Export_Merchandise_Quarterly there is total 1298 Null Values.

##### So First, We will try to fix this. We need to handle these missing values, so that our data is cleaned and Structured. By Filling missing values with Zero in all the tables.

In [47]:
# Filling Null Values with zeroes:
Export_Merchandise_Annual=Export_Merchandise_Annual.fillna(0)
Export_Merchandise_Monthly=Export_Merchandise_Monthly.fillna(0)
Export_Merchandise_Quarterly=Export_Merchandise_Quarterly.fillna(0)

In [48]:
Export_Merchandise_Annual.isnull().sum().sum()

0

In [49]:
Export_Merchandise_Monthly.isnull().sum().sum()

0

In [50]:
Export_Merchandise_Quarterly.isnull().sum().sum()

0

#### Now the Export Merchandise Data is clean and we have replace all the null values with zeros
- All the data are cleaned and do not have any error or null values

### Now let's check for any Duplicate Data in the all the above files - 

In [53]:
Exchange_Rate_monthly.duplicated().sum()
Exchange_Rate_annual.duplicated().sum()

0

In [56]:
CPI_Data_Annual.duplicated().sum()
CPI_Data_Monthly.duplicated().sum()

0

In [57]:
Export_Merchandise_Annual.duplicated().sum()
Export_Merchandise_Monthly.duplicated().sum()
Export_Merchandise_Quarterly.duplicated().sum()

0

#### There is no duplicates value in all the above datas

# Exchange Rate Data Analysis

In [59]:
Exchange_Rate_monthly.describe()

Afghanistan        Angola     Albania  United Arab Emirates  \
count   360.000000  3.600000e+02  360.000000            360.000000   
mean     32.186290  4.751834e+04   96.485436              3.673313   
std      26.305351  1.687534e+05   45.734716              0.001281   
min       0.000000  0.000000e+00    0.000000              3.663136   
25%       0.000000  4.039077e+01   95.418430              3.673000   
50%      47.455885  8.946021e+01  105.073500              3.673000   
75%      50.072937  1.682201e+02  124.450950              3.673045   
max      69.778190  2.179222e+06  176.522100              3.678714   

        Argentina     Armenia  Developing Asia  Antigua and Barbuda  \
count  360.000000  360.000000       360.000000           360.000000   
mean     4.221138  276.020862       800.864416             2.288609   
std      5.793594  228.384080       370.982459             0.962823   
min      0.000000    0.000000       190.359200             0.000000   
25%      0.999950    0.000000       324.826175             2.674643   
50%      2.926575  370.351300       913.325400             2.697505   
75%      4.085869  478.268150      1010.439000             2.700000   
max     38.647830  590.400000      1473.699000             2.730000   

        Australia     Austria  ...  St. Vincent and the Grenadines  \
count  360.000000  360.000000  ...                      360.000000   
mean     1.344120    0.837379  ...                        1.665000   
std      0.231245    0.111153  ...                        1.314563   
min      0.927598    0.634319  ...                        0.000000   
25%      1.241605    0.757501  ...                        0.000000   
50%      1.318294    0.813287  ...                        2.700000   
75%      1.436994    0.892076  ...                        2.700000   
max      1.989637    1.171931  ...                        2.700000   

       Venezuela, RB       Vietnam     Vanuatu  World (WBG members)  \
count     360.000000    360.000000  360.000000           360.000000   
mean     1727.373325  14222.742583   95.902486           344.282931   
std     14855.671846   6992.745617   42.411464           334.569484   
min         0.000000      0.000000    0.000000            76.765460   
25%         0.000000  11011.097500   96.006660           197.004375   
50%         1.598000  15639.790000  109.204700           265.973450   
75%         4.295000  20666.095000  116.369775           425.945475   
max    213716.500000  23323.630000  148.602400          3011.827000   

            Samoa  Yemen, Rep.  South Africa      Zambia    Zimbabwe  
count  360.000000   360.000000    360.000000  360.000000  360.000000  
mean     1.817750   145.659671      7.168952    3.808138   72.016502  
std      1.267106    86.587314      3.407353    3.016914  133.882615  
min      0.000000     0.000000      2.349682    0.000000    0.000000  
25%      0.000000    58.547542      4.361425    1.221001    0.000000  
50%      2.451523   177.966150      6.924533    3.982477    0.000000  
75%      2.687471   214.853775      8.938738    5.026875    0.000000  
max      3.600299   250.353700     16.295760   12.158760  322.355000  

[8 rows x 200 columns]

In [60]:
Exchange_Rate_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Columns: 201 entries, Period to Zimbabwe
dtypes: float64(192), int64(8), object(1)
memory usage: 565.4+ KB


In [61]:
Exchange_Rate_annual.describe()

Period  Afghanistan         Angola     Albania  \
count    30.000000    30.000000      30.000000   30.000000   
mean   2003.500000    33.727701   47587.127773   99.231948   
std       8.803408    26.802241  114758.886901   43.450594   
min    1989.000000     0.000000       0.000000    0.000000   
25%    1996.250000     0.000000      74.018648   98.537425   
50%    2003.500000    47.428790      92.808430  104.928000   
75%    2010.750000    50.306153     227.888350  123.447075   
max    2018.000000    69.363510  516812.700000  152.499600   

       United Arab Emirates   Argenti0     Armenia  Developing Asia  \
count             30.000000  30.000000   30.000000        30.000000   
mean               3.673308   4.296596  285.391613       805.773337   
std                0.000845   5.868982  230.015974       371.197281   
min                3.672184   0.000000    0.000000       213.169500   
25%                3.672992   0.999887    0.000000       337.342025   
50%                3.673000   2.944350  373.080700       917.782250   
75%                3.673203   4.074918  479.845725       990.520100   
max                3.675296  27.165270  578.763000      1394.157000   

       Antigua and Barbuda  Australia  ...  St. Vincent and the Gre0dines  \
count            30.000000  30.000000  ...                      30.000000   
mean              2.333559   1.344591  ...                       1.710000   
std               0.930995   0.224836  ...                       1.323358   
min               0.000000   0.965681  ...                       0.000000   
25%               2.680584   1.266517  ...                       0.000000   
50%               2.696873   1.329737  ...                       2.700000   
75%               2.699744   1.365685  ...                       2.700000   
max               2.700000   1.933352  ...                       2.700000   

       Venezuela, RB        Viet0m     Vanuatu  World (WBG members)  \
count      30.000000     30.000000   30.000000            30.000000   
mean     2589.940503  14435.202333   97.901379           335.445063   
std     14172.708540   6874.977461   41.148145           218.242101   
min         0.000000      0.000000    0.000000            85.265280   
25%         0.000000  11179.995000   97.817770           253.143600   
50%         1.747686  15624.685000  108.663650           270.665700   
75%         4.282214  20273.802500  115.616750           423.500900   
max     77629.490000  22992.540000  145.723000          1107.049000   

           Samoa  Yemen, Rep.  South Africa     Zambia    Zimbabwe  
count  30.000000    30.000000     30.000000  30.000000   30.000000  
mean    1.872398   146.583827      7.198975   3.840597   73.679367  
std     1.236844    86.643212      3.420672   3.037610  136.069115  
min     0.000000     0.000000      2.587400   0.000000    0.000000  
25%     0.429710    77.948250      4.374481   1.215782    0.000000  
50%     2.466643   179.802950      6.997530   3.872197    0.000000  
75%     2.685982   215.002300      8.566305   5.001506    0.000000  
max     3.517049   250.266400     14.706100  10.333910  322.355000  

[8 rows x 201 columns]

In [75]:
# First we need to Rename Period coulmn into the Year
Exchange_Rate_monthly.rename(columns = {'Period':'Year'} , inplace = True)

In [64]:
Exchange_Rate_annual.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Columns: 201 entries, Year to Zimbabwe
dtypes: float64(192), int64(9)
memory usage: 47.2 KB


In [71]:
from datetime import datetime

In [83]:
# Now we need to format the Year column as YYYY-MM-DD__ Format in monthly dataframe .
Date1 = [ ]
for i in range(len(Exchange_Rate_monthly)):
    Date1.append(Exchange_Rate_monthly['Year'][i][0:4])    

# For MM format we need to selected 5th and 6th index as they represent the month number.  
Date2 = [ ]
for i in range(len(Exchange_Rate_monthly)):
    Date2.append(Exchange_Rate_monthly['Year'][i][5:7])    

# Now to add YYYY and MM together we need to create another list
Date3 = [ ]
for i in range(len(Date1)):
    Date3.append(Date2[i]+'-'+Date1[i])
    
def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '01-'
Date3 = prepend(Date3,a)

Date3 = [datetime.strptime(x,'%d-%m-%Y') for x in Date3]

# Find the name of the column by index
n = Exchange_Rate_monthly.columns[0]

# Drop that column
Exchange_Rate_monthly.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
Exchange_Rate_monthly.insert(0,'Year',Date3)

In [84]:
Exchange_Rate_monthly.head()

Year  Afghanistan  Angola  Albania  United Arab Emirates  Argentina  \
0 1988-12-01          0.0     0.0      0.0                 3.675        0.0   
1 1989-01-01          0.0     0.0      0.0                 3.675        0.0   
2 1989-02-01          0.0     0.0      0.0                 3.675        0.0   
3 1989-03-01          0.0     0.0      0.0                 3.675        0.0   
4 1989-04-01          0.0     0.0      0.0                 3.675        0.0   

   Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
0      0.0         190.3592                  0.0   1.164807  ...   
1      0.0         191.0076                  0.0   1.148810  ...   
2      0.0         191.9781                  0.0   1.169614  ...   
3      0.0         193.4470                  0.0   1.223637  ...   
4      0.0         195.0895                  0.0   1.243114  ...   

   St. Vincent and the Grenadines  Venezuela, RB  Vietnam  Vanuatu  \
0                             0.0            0.0      0.0      0.0   
1                             0.0            0.0      0.0      0.0   
2                             0.0            0.0      0.0      0.0   
3                             0.0            0.0      0.0      0.0   
4                             0.0            0.0      0.0      0.0   

   World (WBG members)  Samoa  Yemen, Rep.  South Africa  Zambia  Zimbabwe  
0             76.76546    0.0          0.0      2.349682     0.0       0.0  
1             77.67523    0.0          0.0      2.385500     0.0       0.0  
2             77.53630    0.0          0.0      2.458750     0.0       0.0  
3             78.12339    0.0          0.0      2.539391     0.0       0.0  
4             78.43185    0.0          0.0      2.548200     0.0       0.0  

[5 rows x 201 columns]

In [100]:
Exchange_Rate_annual.dtypes

Period                    int64
Afghanistan             float64
Angola                  float64
Albania                 float64
United Arab Emirates    float64
                         ...   
Samoa                   float64
Yemen, Rep.             float64
South Africa            float64
Zambia                  float64
Zimbabwe                float64
Length: 201, dtype: object

In [102]:
Exchange_Rate_annual.head()

Period  Afghanistan    Angola   Albania  United Arab Emirates  Argenti0  \
0    1989          0.0     0.000    0.0000              3.675000  0.000000   
1    1990          0.0     0.000    0.0000              3.674964  0.000000   
2    1991          0.0     0.000    0.0000              3.675188  0.000000   
3    1992          0.0     0.000    0.0000              3.673874  0.991757   
4    1993          0.0  4903.561  110.0993              3.675296  0.998979   

   Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
0      0.0         213.1695              0.00000   1.262848  ...   
1      0.0         242.0275              0.00000   1.279588  ...   
2      0.0         282.4442              0.00000   1.283241  ...   
3      0.0         290.6883              0.00000   1.361321  ...   
4      0.0         299.6200              2.69682   1.471305  ...   

   St. Vincent and the Gre0dines  Venezuela, RB    Viet0m   Vanuatu  \
0                            0.0            0.0      0.00    0.0000   
1                            0.0            0.0      0.00    0.0000   
2                            0.0            0.0      0.00    0.0000   
3                            0.0            0.0      0.00    0.0000   
4                            0.0            0.0  10698.57  122.3616   

   World (WBG members)  Samoa  Yemen, Rep.  South Africa  Zambia  Zimbabwe  
0             85.26528    0.0      0.00000      2.621661     0.0       0.0  
1             90.01011    0.0      0.00000      2.587400     0.0       0.0  
2             89.66550    0.0      0.00000      2.759186     0.0       0.0  
3             91.20521    0.0      0.00000      2.853479     0.0       0.0  
4             91.83828    0.0     16.48504      3.269305     0.0       0.0  

[5 rows x 201 columns]

In [103]:
# Changing Year Column to Datetime
 
data1 = [i for i in Exchange_Rate_annual['Period']]

    
def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '31-12-'
data1 = prepend(data1,a)

data1 = [datetime.strptime(x,'%d-%m-%Y') for x in data1]

# Find the name of the column by index
n = Exchange_Rate_annual.columns[0]

# Drop that column
Exchange_Rate_annual.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
Exchange_Rate_annual.insert(0,'Year',data1)

In [104]:
Exchange_Rate_annual.head()

Year  Afghanistan    Angola   Albania  United Arab Emirates  Argenti0  \
0 1989-12-31          0.0     0.000    0.0000              3.675000  0.000000   
1 1990-12-31          0.0     0.000    0.0000              3.674964  0.000000   
2 1991-12-31          0.0     0.000    0.0000              3.675188  0.000000   
3 1992-12-31          0.0     0.000    0.0000              3.673874  0.991757   
4 1993-12-31          0.0  4903.561  110.0993              3.675296  0.998979   

   Armenia  Developing Asia  Antigua and Barbuda  Australia  ...  \
0      0.0         213.1695              0.00000   1.262848  ...   
1      0.0         242.0275              0.00000   1.279588  ...   
2      0.0         282.4442              0.00000   1.283241  ...   
3      0.0         290.6883              0.00000   1.361321  ...   
4      0.0         299.6200              2.69682   1.471305  ...   

   St. Vincent and the Gre0dines  Venezuela, RB    Viet0m   Vanuatu  \
0                            0.0            0.0      0.00    0.0000   
1                            0.0            0.0      0.00    0.0000   
2                            0.0            0.0      0.00    0.0000   
3                            0.0            0.0      0.00    0.0000   
4                            0.0            0.0  10698.57  122.3616   

   World (WBG members)  Samoa  Yemen, Rep.  South Africa  Zambia  Zimbabwe  
0             85.26528    0.0      0.00000      2.621661     0.0       0.0  
1             90.01011    0.0      0.00000      2.587400     0.0       0.0  
2             89.66550    0.0      0.00000      2.759186     0.0       0.0  
3             91.20521    0.0      0.00000      2.853479     0.0       0.0  
4             91.83828    0.0     16.48504      3.269305     0.0       0.0  

[5 rows x 201 columns]

# CPI Data Analysis

In [87]:
CPI_Data_Monthly.dtypes

Period                   object
Afghanistan             float64
Albania                 float64
United Arab Emirates    float64
Armenia                 float64
                         ...   
World (WBG members)     float64
Samoa                   float64
Yemen, Rep.             float64
South Africa            float64
Zambia                  float64
Length: 173, dtype: object

In [94]:
# conversion into YYYY-MM-DD Format.

# For YYYY we have selected from 0th index to 4th index    
data1 = [ ]
for i in range(len(CPI_Data_Monthly)):
    data1.append(CPI_Data_Monthly['Period'][i][0:4])    

# For MM format we have selected 5th and 6th index as they represent the month number.  
data2 = [ ]
for i in range(len(CPI_Data_Monthly)):
    data2.append(CPI_Data_Monthly['Period'][i][5:7])    

# Now to add YYYY and MM together we create another list
data3 = [ ]
for i in range(len(data1)):
    data3.append(data2[i]+'-'+data1[i])
    
def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '01-'
data3 = prepend(data3,a)

data3 = [datetime.strptime(x,'%d-%m-%Y') for x in data3]

# Find the name of the column by index
n = CPI_Data_Monthly.columns[0]

# Drop that column
CPI_Data_Monthly.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
CPI_Data_Monthly.insert(0,'Year',data3)

In [95]:
CPI_Data_Monthly.head()

Year  Afghanistan  Albania  United Arab Emirates  Armenia  \
0 1988-12-01          0.0      0.0                   0.0      0.0   
1 1989-01-01          0.0      0.0                   0.0      0.0   
2 1989-02-01          0.0      0.0                   0.0      0.0   
3 1989-03-01          0.0      0.0                   0.0      0.0   
4 1989-04-01          0.0      0.0                   0.0      0.0   

   Developing Asia  Antigua and Barbuda   Austria  Azerbaijan  Burundi  ...  \
0         33.23211                  0.0  63.07520         0.0      0.0  ...   
1         34.38592                  0.0  63.63636         0.0      0.0  ...   
2         34.60887                  0.0  63.74860         0.0      0.0  ...   
3         35.11426                  0.0  63.86083         0.0      0.0  ...   
4         35.52607                  0.0  64.08530         0.0      0.0  ...   

    Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
0  0.720415       55.61017                             0.0       0.199333   
1  0.752198       55.88706                             0.0       0.201556   
2  0.794576       56.11781                             0.0       0.207910   
3  0.858142       56.44086                             0.0       0.252126   
4  0.900519       56.81005                             0.0       0.286180   

   Vietnam  World (WBG members)  Samoa  Yemen, Rep.  South Africa    Zambia  
0      0.0             48.23800    0.0          0.0      19.62482  0.095229  
1      0.0             48.59190    0.0          0.0      19.91342  0.095229  
2      0.0             48.75023    0.0          0.0      20.20202  0.095229  
3      0.0             49.03835    0.0          0.0      20.49062  0.095229  
4      0.0             49.49379    0.0          0.0      20.77922  0.095229  

[5 rows x 173 columns]

In [97]:
CPI_Data_Annual.dtypes

Year                      int64
Afghanistan             float64
Albania                 float64
United Arab Emirates    float64
Armenia                 float64
                         ...   
World (WBG members)     float64
Samoa                   float64
Yemen, Rep.             float64
South Africa            float64
Zambia                  float64
Length: 173, dtype: object

In [98]:
# changing Year column to datetime
l = [i for i in CPI_Data_Annual['Year']]
def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '31-12-'
l = prepend(l,a)      

from datetime import datetime

l = [datetime.strptime(x,'%d-%m-%Y') for x in l]

# Find the name of the column by index
n = CPI_Data_Annual.columns[0]

# Drop that column
CPI_Data_Annual.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
CPI_Data_Annual.insert(0,'Year',l)

In [99]:
CPI_Data_Annual.head()

Year  Afghanistan  Albania  United Arab Emirates   Armenia  \
0 1989-12-31          0.0      0.0                   0.0  0.000000   
1 1990-12-31          0.0      0.0                   0.0  0.000000   
2 1991-12-31          0.0      0.0                   0.0  0.000000   
3 1992-12-31          0.0      0.0                   0.0  0.019203   
4 1993-12-31          0.0      0.0                   0.0  0.326774   

   Developing Asia  Antigua and Barbuda   Austria  Azerbaijan   Burundi  ...  \
0         35.24378                  0.0  64.49682         0.0   0.00000  ...   
1         36.50166                  0.0  66.60120         0.0   0.00000  ...   
2         38.06775                  0.0  68.85522         0.0   0.00000  ...   
3         40.46016                  0.0  71.59559         0.0   0.00000  ...   
4         45.49808                  0.0  74.24242         0.0  13.03742  ...   

     Uruguay  United States  St. Vincent and the Grenadines  Venezuela, RB  \
0   1.036480       57.21002                             0.0       0.301840   
1   2.202740       60.29819                             0.0       0.424556   
2   4.446092       62.85179                             0.0       0.569776   
3   7.488435       64.75546                             0.0       0.748814   
4  11.539000       66.66682                             0.0       1.034275   

   Vietnam  World (WBG members)     Samoa  Yemen, Rep.  South Africa    Zambia  
0      0.0             49.71672   0.00000          0.0      21.32035  0.119036  
1      0.0             52.16386   0.00000          0.0      24.35065  0.222201  
2      0.0             54.75251   0.00000          0.0      28.05435  0.452338  
3      0.0             56.90573   0.00000          0.0      31.99856  1.174491  
4      0.0             59.44683  46.15621          0.0      35.08899  3.325080  

[5 rows x 173 columns]

# Export Merchandise Data Analysis

In [105]:
Export_Merchandise_Annual.dtypes

Year                                      int64
Argentina                               float64
Developing Asia                         float64
Australia                               float64
Belgium                                 float64
Brazil                                  float64
Canada                                  float64
Chile                                   float64
China                                   float64
Czech Republic                          float64
Germany                                 float64
Developing Countries                    float64
Denmark                                 float64
East Asia & Pacific developing          float64
Europe & Central Asia developing        float64
Spain                                   float64
Estonia                                 float64
Finland                                 float64
France                                  float64
United Kingdom                          float64
High Income Countries                   

In [107]:
# changing Year column to datetime
l = [i for i in Export_Merchandise_Annual['Year']]
def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '31-12-'
l = prepend(l,a)      

l = [datetime.strptime(x,'%d-%m-%Y') for x in l]

# Find the name of the column by index
n = Export_Merchandise_Annual.columns[0]

# Drop that column
Export_Merchandise_Annual.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
Export_Merchandise_Annual.insert(0,'Year',l)

In [108]:
Export_Merchandise_Annual.head()

Year  Argentina  Developing Asia  Australia   Belgium    Brazil  \
0 1991-12-31   22191.58              0.0   80560.51       0.0  53953.57   
1 1992-12-31   22254.20              0.0   85549.90       0.0  63269.99   
2 1993-12-31   24153.33              0.0   90912.32       0.0  73589.75   
3 1994-12-31   28583.86              0.0   96492.75       0.0  75080.52   
4 1995-12-31   35332.53              0.0   99382.24  296008.8  70497.63   

   Canada  Chile  China  Czech Republic  ...  Singapore  \
0     0.0    0.0    0.0             0.0  ...   47745.83   
1     0.0    0.0    0.0             0.0  ...   51734.70   
2     0.0    0.0    0.0             0.0  ...   61247.96   
3     0.0    0.0    0.0             0.0  ...   78650.06   
4     0.0    0.0    0.0             0.0  ...   91012.18   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand     Turkey  \
0                            0.0       0.0  66384.82  44867.88   51.61588   
1                            0.0       0.0  66534.56  50383.36   89.88207   
2                            0.0       0.0  72278.16  57262.15  156.75590   
3                            0.0       0.0  83427.53  66941.26  521.16430   
4                            0.0       0.0  90334.67  76568.04  837.45090   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       64603.98     0.0       528704.3                  0.0  
1       68509.85     0.0       561466.6                  0.0  
2       71342.15     0.0       579401.9                  0.0  
3       77694.99     0.0       625109.1            4948934.0  
4       87338.68     0.0       679022.4            5375153.0  

[5 rows x 55 columns]

In [112]:
Export_Merchandise_Monthly.head()

Period  Argentina  Developing Asia  Australia  Belgium    Brazil  Canada  \
0  1991M01   1265.303         3313.754   5502.346      0.0  4896.404     0.0   
1  1991M02   1518.856            0.000   5708.852      0.0  3887.668     0.0   
2  1991M03   1632.372            0.000   6890.402      0.0  4384.693     0.0   
3  1991M04   1884.038            0.000   6524.655      0.0  5119.613     0.0   
4  1991M05   2302.835            0.000   7145.694      0.0  4917.469     0.0   

   Chile  China  Czech Republic  ...  Singapore  \
0    0.0    0.0             0.0  ...   4014.566   
1    0.0    0.0             0.0  ...   3175.083   
2    0.0    0.0             0.0  ...   4062.258   
3    0.0    0.0             0.0  ...   3896.417   
4    0.0    0.0             0.0  ...   3908.034   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand    Turkey  \
0                            0.0       0.0  5493.817  3313.754  2.651407   
1                            0.0       0.0  5402.757  3152.839  2.588754   
2                            0.0       0.0  5898.220  4010.966  3.935234   
3                            0.0       0.0  6062.707  3425.034  3.178725   
4                            0.0       0.0  5544.209  3584.901  3.911652   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       5268.258     0.0       41353.47             168083.9  
1       4161.086     0.0       40909.06                  0.0  
2       4857.343     0.0       46101.14                  0.0  
3       4778.434     0.0       45152.55                  0.0  
4       5758.457     0.0       45104.24                  0.0  

[5 rows x 55 columns]

In [109]:
Export_Merchandise_Monthly.dtypes

Period                                   object
Argentina                               float64
Developing Asia                         float64
Australia                               float64
Belgium                                 float64
Brazil                                  float64
Canada                                  float64
Chile                                   float64
China                                   float64
Czech Republic                          float64
Germany                                 float64
Developing Countries                    float64
Denmark                                 float64
East Asia & Pacific developing          float64
Europe & Central Asia developing        float64
Spain                                   float64
Estonia                                 float64
Finland                                 float64
France                                  float64
United Kingdom                          float64
High Income Countries                   

In [114]:
# changing Year column to datetime
date1 = [ ]
for i in range(len(Export_Merchandise_Monthly)):
    date1.append(Export_Merchandise_Monthly['Period'][i][0:4])    

# For MM format we have selected 5th and 6th index as they represent the month number.  
date2 = [ ]
for i in range(len(Export_Merchandise_Monthly)):
    date2.append(Export_Merchandise_Monthly['Period'][i][5:7])    

# Now to add YYYY and MM together we create another list
date3 = [ ]
for i in range(len(date1)):
    date3.append(date2[i]+'-'+date1[i])

def prepend(list, str):
 
    # Using format()
    str += '{0}'
    list = [str.format(i) for i in list]
    return(list)
a = '01-'
date3 = prepend(date3,a)      

date3 = [datetime.strptime(x,'%d-%m-%Y') for x in date3]

# Find the name of the column by index
n = Export_Merchandise_Monthly.columns[0]

# Drop that column
Export_Merchandise_Monthly.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place
Export_Merchandise_Monthly.insert(0,'Year',date3)

In [115]:
Export_Merchandise_Monthly.head()

Year  Argentina  Developing Asia  Australia  Belgium    Brazil  \
0 1991-01-01   1265.303         3313.754   5502.346      0.0  4896.404   
1 1991-02-01   1518.856            0.000   5708.852      0.0  3887.668   
2 1991-03-01   1632.372            0.000   6890.402      0.0  4384.693   
3 1991-04-01   1884.038            0.000   6524.655      0.0  5119.613   
4 1991-05-01   2302.835            0.000   7145.694      0.0  4917.469   

   Canada  Chile  China  Czech Republic  ...  Singapore  \
0     0.0    0.0    0.0             0.0  ...   4014.566   
1     0.0    0.0    0.0             0.0  ...   3175.083   
2     0.0    0.0    0.0             0.0  ...   4062.258   
3     0.0    0.0    0.0             0.0  ...   3896.417   
4     0.0    0.0    0.0             0.0  ...   3908.034   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand    Turkey  \
0                            0.0       0.0  5493.817  3313.754  2.651407   
1                            0.0       0.0  5402.757  3152.839  2.588754   
2                            0.0       0.0  5898.220  4010.966  3.935234   
3                            0.0       0.0  6062.707  3425.034  3.178725   
4                            0.0       0.0  5544.209  3584.901  3.911652   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       5268.258     0.0       41353.47             168083.9  
1       4161.086     0.0       40909.06                  0.0  
2       4857.343     0.0       46101.14                  0.0  
3       4778.434     0.0       45152.55                  0.0  
4       5758.457     0.0       45104.24                  0.0  

[5 rows x 55 columns]

In [116]:
Export_Merchandise_Quarterly.dtypes

Period                                   object
Argentina                               float64
Developing Asia                         float64
Australia                               float64
Belgium                                 float64
Brazil                                  float64
Canada                                  float64
Chile                                   float64
China                                   float64
Czech Republic                          float64
Germany                                 float64
Developing Countries                    float64
Denmark                                 float64
East Asia & Pacific developing          float64
Europe & Central Asia developing        float64
Spain                                   float64
Estonia                                 float64
Finland                                 float64
France                                  float64
United Kingdom                          float64
High Income Countries                   

In [120]:
# For YYYY we have selected from 0th index to 4th index    
l1 = [ ]
for i in range(len(Export_Merchandise_Quarterly)):
    l1.append(Export_Merchandise_Quarterly['Period'][i][0:4])    

# For QQ format we have selected 4th and 5th index as they represent the month number.  
l2 = [ ]
for i in range(len(Export_Merchandise_Quarterly)):
    l2.append(Export_Merchandise_Quarterly['Period'][i][4:6])    

# Now to add YYYY and QQ together we create another list
l3 = [ ]
for i in range(len(l1)):
    l3.append(l1[i]+'-'+l2[i])
    
l3 = pd.to_datetime(l3)

# Find the name of the column by index
n = Export_Merchandise_Quarterly.columns[0]

# Drop that column
Export_Merchandise_Quarterly.drop(n, axis = 1, inplace = True)

# Put whatever series you want in its place

Export_Merchandise_Quarterly.insert(0,'Year',l3)

In [121]:
Export_Merchandise_Quarterly.head()

Year  Argentina  Developing Asia  Australia  Belgium    Brazil  \
0 1991-01-01   4416.531              0.0   18101.60      0.0  13168.76   
1 1991-04-01   6318.661              0.0   20387.41      0.0  14826.59   
2 1991-07-01   6371.975              0.0   20408.96      0.0  12690.13   
3 1991-10-01   5084.412              0.0   21662.54      0.0  13268.09   
4 1992-01-01   4724.392              0.0   19736.18      0.0  13636.11   

   Canada  Chile  China  Czech Republic  ...  Singapore  \
0     0.0    0.0    0.0             0.0  ...   11251.91   
1     0.0    0.0    0.0             0.0  ...   12001.44   
2     0.0    0.0    0.0             0.0  ...   12429.65   
3     0.0    0.0    0.0             0.0  ...   12062.84   
4     0.0    0.0    0.0             0.0  ...   11953.98   

   Sub-Saharan Africa developing  Slovakia    Sweden  Thailand     Turkey  \
0                            0.0       0.0  16794.79  10477.56   9.175394   
1                            0.0       0.0  17536.09  10505.70  10.675940   
2                            0.0       0.0  14657.51  12132.41  13.381980   
3                            0.0       0.0  17396.43  11752.21  18.382570   
4                            0.0       0.0  17557.24  11931.11  17.588660   

   Taiwan, China  Uganda  United States  World (WBG members)  
0       14286.69     0.0       128363.7                  0.0  
1       16374.93     0.0       134818.8                  0.0  
2       17147.92     0.0       127887.5                  0.0  
3       16794.44     0.0       137634.3                  0.0  
4       16101.55     0.0       139734.4                  0.0  

[5 rows x 55 columns]

In [122]:
Export_Merchandise_Quarterly.describe()

Argentina  Developing Asia     Australia       Belgium  \
count    111.000000       111.000000    111.000000    111.000000   
mean   12880.512937    253912.501802  37454.481532  73821.483604   
std     4027.077206    284850.980625  11447.790935  32173.940626   
min        0.000000         0.000000      0.000000      0.000000   
25%    10791.245000         0.000000  29830.860000  75477.225000   
50%    13738.760000         0.000000  36487.320000  88300.780000   
75%    16095.565000    549909.900000  45680.885000  92029.185000   
max    20018.850000    735916.600000  60303.700000  98223.310000   

             Brazil      Canada         Chile          China  Czech Republic  \
count    111.000000  111.000000    111.000000     111.000000      111.000000   
mean   35150.514955   84.081887   9444.140991  187474.458559    23022.802973   
std    14871.974009   46.275518   8510.739951  209170.797746    18278.579328   
min        0.000000    0.000000      0.000000       0.000000        0.000000   
25%    20170.815000   88.614225      0.000000       0.000000        0.000000   
50%    38919.610000  107.980400  15192.980000       0.000000    22197.140000   
75%    48712.660000  113.214100  17292.965000  396189.050000    38321.345000   
max    61736.420000  126.261700  19218.730000  552142.000000    57857.500000   

             Germany  ...      Singapore  Sub-Saharan Africa developing  \
count     111.000000  ...     111.000000                     111.000000   
mean   268471.486486  ...   57210.344144                     395.739287   
std     99671.950950  ...   32772.906646                     426.992331   
min         0.000000  ...       0.000000                       0.000000   
25%    179413.250000  ...   27669.940000                       0.000000   
50%    274460.500000  ...   58324.250000                     317.129500   
75%    358342.550000  ...   90532.345000                     669.016100   
max    435491.800000  ...  110839.100000                    1569.471000   

         Slovakia        Sweden      Thailand        Turkey  Taiwan, China  \
count  111.000000    111.000000    111.000000     111.00000     111.000000   
mean     7.401956  33667.630811  34192.664234   25849.78212   44559.580631   
std     10.615051   9817.347991  15718.175496   29621.45896   21170.064235   
min      0.000000      0.000000      0.000000       0.00000       0.000000   
25%      0.000000  26782.775000  18692.775000    1143.85700   24490.530000   
50%      0.000000  35175.740000  33880.880000   19387.62000   43082.150000   
75%     18.487160  41971.840000  50840.740000   33834.08500   65832.955000   
max     28.776680  48239.380000  58368.110000  126211.00000   83571.290000   

            Uganda  United States  World (WBG members)  
count   111.000000     111.000000         1.110000e+02  
mean    397.825786  256836.846847         2.185167e+06  
std     425.604576   83534.042515         1.005887e+06  
min       0.000000       0.000000         0.000000e+00  
25%       0.000000  201071.350000         1.580191e+06  
50%     317.129500  243749.500000         2.185551e+06  
75%     669.016100  335515.450000         3.032770e+06  
max    1569.471000  404231.900000         3.810925e+06  

[8 rows x 54 columns]